core/src/test/java/org/apache/iceberg/TestRemoveSnapshots.java

```diff
@@ -1772,6 +1775,24 @@ public void testNoSchemasOrSpecsToRemove() {
                         .anyMatch(u -> u instanceof MetadataUpdate.RemoveSchemas)));
   }
 
+  @TestTemplate
+  public void testCustomWorkerPool() {
+    ExecutorService executor = Executors.newSingleThreadExecutor();
+    // shutdown the executor, cleanExpiredSnapshots should fail
+    executor.shutdownNow();
+    RemoveSnapshots removeSnapshots = (RemoveSnapshots) removeSnapshots(table).planWith(executor);
+
+    table.newAppend().appendFile(FILE_A).commit();
+    table.newAppend().appendFile(FILE_A).commit();
+
+    long tAfterCommits = waitUntilAfter(table.currentSnapshot().timestampMillis());
+    assertThrows(
```

Comment:
Please use junit5: assertThatThrownBy
We are in the process of migrating, and still both versions in the classpath, but we should use junit5

Reply:
ack